### PR TITLE
Update Puppet module versions

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: puppet/Puppetfile [hunk 1]
```diff
@@ -10,41 +10,40 @@ mod 'puppet/augeasproviders_shellvar', '6.0.0'
...
 mod 'puppetlabs/mount_providers', '2.0.1'
-mod 'puppetlabs/ntp', '9.2.2'
-mod 'puppetlabs/postgresql', '8.2.1'
-# released version doesn't allow puppetlabs/concat 7.x
-mod 'puppetlabs/rsync', :git => 'https://github.com/ekohl/puppetlabs-rsync', :branch => 'allow-concat-7'
-mod 'puppetlabs/stdlib', '8.6.0'
+mod 'puppetlabs/ntp', '10.1.0'
+mod 'puppetlabs/postgresql', '10.0.2'
+# released version doesn't allow puppetlabs/concat 9.x & puppetlabs/stdlib 9.x
+mod 'puppetlabs/rsync', :git => 'https://github.com/ekohl/puppetlabs-rsync', :branch => 'compatible-with-latest'
```
Comment (ekohl) [inline puppet/Puppetfile:34]: https://github.com/puppetlabs/puppetlabs-rsync/pull/151

File: puppet/Puppetfile [hunk 2]
```diff
@@ -10,41 +10,40 @@ mod 'puppet/augeasproviders_shellvar', '6.0.0'
...
 mod 'puppetlabs/vcsrepo', '6.1.0'
 mod 'puppetlabs/xinetd', '3.4.1'
 mod 'richardc/datacat', '0.6.2'
 mod 'saz/sudo', '8.0.0'
-mod 'theforeman/foreman', '23.2.0'
-mod 'theforeman/foreman_proxy', '25.1.0'
-mod 'theforeman/git', '7.2.0'
+mod 'theforeman/foreman', '24.1.0'
+mod 'theforeman/foreman_proxy', '25.2.0'
 mod 'theforeman/motd', '1.0.0'
```
Comment (ekohl) [inline puppet/Puppetfile:42]: Current failure:
```console
$ be ./check_dependencies 
Unable to satisfy theforeman/motd
Investigating version 1.0.0
  Dependency puppetlabs/stdlib (>= 4.25.0 < 9.0.0) can't be satified
    Available versions: 9.4.1
```

Comment (ekohl) [reply puppet/Puppetfile:42]: https://github.com/theforeman/puppet-motd/pull/17

**Comment by ekohl** [comment]
Currently fails on https://github.com/puppetlabs/puppetlabs-java/pull/569 so I wonder if we should drop Debian 12 from the supported versions for now.

**Comment by evgeni** [comment]
> Currently fails on https://github.com/puppetlabs/puppetlabs-java/pull/569 so I wonder if we should drop Debian 12 from the supported versions for now.

I'd be fine with that, we don't have any D12 nodes right now.

**Review by evgeni — APPROVED** [review]
one way to find out…

**Pull request merged**